gen/src/writer/ucd/bidi.rs

```diff
@@ -27,16 +27,10 @@ use generate::tables::{ToRangeCharSet, ToRangeCharTable};
 struct BidiClassData<'a>(BTreeMap<char, &'a str>);
 
 impl<'a> BidiClassData<'a> {
-    fn emit<P: AsRef<Path>>(&self, dir: P) -> io::Result<()> {
+    fn emit(&self, dir: &PathBuf) {
```

Comment:
If we're going to make these concrete types, we should use `&Path` instead of `&PathBuf`. `Path` is the slice type, and `&PathBuf` coerces to `&Path`.

Reply:
Right. I'll fix them.